src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala

```diff
@@ -1,418 +1,517 @@
-import it.pagopa.interop.commons.utils.TypeConversions.{EitherOps, OptionOps, StringOps}
+import it.pagopa.interop.commons.utils.AkkaUtils.getClaimFuture
+import it.pagopa.interop.commons.utils.ORGANIZATION_ID_CLAIM
+import it.pagopa.interop.commons.utils.OpenapiUtils.parseArrayParameters
+import it.pagopa.interop.commons.utils.TypeConversions.{EitherOps, StringOps}
+import it.pagopa.interop.tenantmanagement.client.{model => TenantManagement}
 
+import java.util.UUID
 import scala.concurrent.{ExecutionContext, Future}
-import scala.util.{Failure, Success}
+import scala.util.Success
 
 final case class AgreementApiServiceImpl(
   agreementManagementService: AgreementManagementService,
   catalogManagementService: CatalogManagementService,
-  partyManagementService: PartyManagementService,
+  tenantManagementService: TenantManagementService,
   attributeManagementService: AttributeManagementService,
-  authorizationManagementService: AuthorizationManagementService,
-  jwtReader: JWTReader
+  authorizationManagementService: AuthorizationManagementService
 )(implicit ec: ExecutionContext)
     extends AgreementApiService {
 
-  val logger: LoggerTakingImplicit[ContextFieldsToLog] =
+  implicit val logger: LoggerTakingImplicit[ContextFieldsToLog] =
     Logger.takingImplicit[ContextFieldsToLog](this.getClass)
 
-  override def activateAgreement(agreementId: String, partyId: String)(implicit
+  override def createAgreement(payload: AgreementPayload)(implicit
     contexts: Seq[(String, String)],
-    toEntityMarshallerProblem: ToEntityMarshaller[Problem]
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem],
+    toEntityMarshallerAgreement: ToEntityMarshaller[Agreement]
   ): Route = authorize(ADMIN_ROLE) {
-    logger.info("Activating agreement {}", agreementId)
+    logger.info(s"Creating agreement $payload")
     val result = for {
-      agreement             <- agreementManagementService.getAgreementById(agreementId)
-      _                     <- verifyAgreementActivationEligibility(agreement)
-      consumerAttributes    <- partyManagementService.getPartyAttributes(agreement.consumerId)
-      consumerAttributesIds <- Future.traverse(consumerAttributes)(a =>
-        attributeManagementService.getAttributeByOriginAndCode(a.origin, a.code).map(_.id)
+      requesterOrgId <- getRequesterOrganizationId(contexts)
+      eService       <- catalogManagementService.getEServiceById(payload.eserviceId)
+      _              <- CatalogManagementService.validateCreationOnDescriptor(eService, payload.descriptorId)
+      _              <- verifyCreationConflictingAgreements(eService.producerId, requesterOrgId, payload)
+      consumer       <- tenantManagementService.getTenant(requesterOrgId)
+      _              <- validateCertifiedAttributes(eService, consumer)
+      agreement      <- agreementManagementService.createAgreement(
+        eService.producerId,
+        requesterOrgId,
+        payload.eserviceId,
+        payload.descriptorId
```

Comment:
I have a couple of considerations: 1) having a bunch of checks that return `Future[Unit]` performed one after another might be worth refactoring in a single function (I can't tell if `CatalogManagementService.validateCreationOnDescriptor(eService, payload.descriptorId)` and `       _              <- verifyCreationConflictingAgreements(eService.producerId, requesterOrgId, payload)`) might be joined but it's worth considering it,
2) orthogonal calls might be performed together via `.zip`, like `tenantManagementService.getTenant(requesterOrgId)` and `catalogManagementService.getEServiceById(payload.eserviceId)` unless there's the specific need to make them in a particular sequence.